cmd/ketch/framework_remove.go

```diff
@@ -127,3 +131,23 @@ func removeNamespace(ctx context.Context, cfg config, framework *ketchv1.Framewo
 
 	return nil
 }
+
+// pruneRemovedAppsFromStatus lists apps and removes any apps that are not present on the system from a framework's status.
+func pruneRemovedAppsFromStatus(ctx context.Context, cfg config, framework ketchv1.Framework) error {
+	var apps ketchv1.AppList
+	if err := cfg.Client().List(ctx, &apps); err != nil {
+		return fmt.Errorf("failed to list framework apps: %w", err)
+	}
+	var updatedApps []string
+	for _, frameworkApp := range framework.Status.Apps {
+		for _, app := range apps.Items {
+			if app.Name == frameworkApp {
+				updatedApps = append(updatedApps, frameworkApp)
+			}
+		}
+	}
+	patchedFramework := framework
+	patchedFramework.Status.Apps = updatedApps
+	patch := client.MergeFrom(&framework)
+	return cfg.Client().Status().Patch(ctx, &patchedFramework, patch)
```

Comment:
Is there a chance to get `the object has been modified; please apply your changes to the latest version and try again`? retry.RetryOnConflict()?